ktor-server/jvm/test/io/ktor/tests/plugins/api/ServerPluginTest.kt

```diff
@@ -294,4 +295,65 @@ class ServerPluginTest {
             )
         }
     }
+
+    @Test
+    fun `test routing scoped install`() {
+        class Config(var data: String)
+
+        val plugin = createRoutingScopedPlugin("F", { Config("default") }) {
+            onCall {
+                context.call.respond(pluginConfig.data)
+            }
+        }
+
+        withTestApplication {
+            application.install(plugin)
```

Comment:
More of a design question.
As I see here, `RoutingScopedPlugin` can be still installed to an application.
And the only difference with a normal plugin is that it overrides settings when installed on subroutes.

1. Doesn't it make sense to make all plugins routing-scoped?
2. From a user perspective, when I'm about to create a plugin, how should I choose: should it be routing-scoped or not. Does it make sense not to make it routing-scoped for some reason?
3. Do we have routing-scoped plugins among our out-of-the-box plugins?

Reply:
Please read the design doc because it answers your questions https://youtrack.jetbrains.com/issue/KTOR-928
TL;DR

The main difference is that routing scoped plugins can access their config only inside call context.
1. No, some plugins can't be routing scoped. Imagine a plugin that schedules some cron jon on application start. It has nothing to do with routing.
2. Usually it's clear when you look at what is the purpose of the plugin.
3. Not yet, but we will default plugins if they can be routing scoped

Reply:
Thanks!
I would personally add some example and a brief explanation to KDoc because users may ask the same questions when writing code. 
cc: @andreyaksenov ?